python/grass/jupyter/timeseries.py

```diff
@@ -0,0 +1,274 @@
+        )
+
+    # Parse string
+    new_rows = [row.split("|") for row in rows]
+    new_array = [list(row) for row in zip(*new_rows)]
+
+    # Collect layer name and start time
+    for column in new_array:
+        if column[0] == "name":
+            names = column[1:]
+        if column[0] == "start_time":
+            dates = column[1:]
+
+    # For variable timestep datasets, fill in None values with
+    # previous time step value. If first time step is missing data,
+    # use the next non-None value
+    for i, name in enumerate(names):
+        if name == "None":
+            if i > 0:
+                names[i] = names[i - 1]
+            else:
+                search_count = 1
+                while name[i + search_count] == "None":
+                    search_count += 1
+                names[i] = name[i + 1]
+        else:
+            pass
+    return names, dates
+
+
+class TimeSeries:
+    """Creates visualizations of time-space raster and vector datasets in Jupyter
+    Notebooks.
+
+    Basic usage::
+    >>> img = TimeSeries("series_name")
+    >>> img.d_legend() #Add legend
+    >>> img.render_layers() #Render Layers
+    >>> img.time_slider() #Create TimeSlider
+    >>> img.animate()
+
+    This class of grass.jupyter is experimental and under development. The API can
+    change at anytime.
+    """
+
+    def __init__(self, timeseries, etype="strds", basemap=None, overlay=None):
+        """Creates an instance of the TimeSeries visualizations class.
+
+        :param str timeseries: name of space-time dataset
+        :param str etype: element type, strds (space-time raster dataset)
+                          or stvds (space-time vector dataset)
+        :param str basemap: name of raster to use as basemap/background for visuals
+        :param str overlay: name of vector to add to visuals
+        """
+        self.timeseries = timeseries
+        self._legend = False
+        self._etype = etype  # element type, borrowing convention from tgis
+        self._renderlist = []
+        self._legend_kwargs = None
+        # self._filenames = []
+        # self._file_date_dict = {}
+        self._date_name_dict = {}
+
+        # Currently does not support multiple basemaps or overlays
+        # (i.e. if you wanted to have two vectors rendered, like
+        # roads and streams, this isn't possible - you can only have one
+        # We should be put a better method here someday
+        self.basemap = basemap
+        self.overlay = overlay
+
+        # Check that map is time space dataset
+        test = gs.read_command("t.list", where=f"name='{timeseries}'")
+        if not test:
+            raise NameError(
+                _(
+                    "Could not find space time raster or vector " "dataset named {}"
+                ).format(timeseries)
+            )
+
+        # Create a temporary directory for our PNG images
+        self._tmpdir = (
+            # pylint: disable=consider-using-with
```

Comment:
Copy the comment from the renderers about using finalize for cleanup to provide reasoning for the pylint: disable.